### PR TITLE
Task 6: redux

### PR DESCRIPTION
**Pull request by DariaGunina**

### CODE REVIEW

File: src/components/Header/Header.tsx [hunk 1]
```diff
@@ -19,11 +24,17 @@ const BackgroundStyle = {
...
+
+    const onSearch = () => getMovies(dispatch, {search, sort, filter});
+
     const [isOpenModal, setIsOpenModal] = React.useState(false);
     const {activeMovie, setActiveMovie} = React.useContext(MovieContext);
+    const {moviesList} = useSelector<RootReducer, MovieState>((state) => state.movies);
 
     const getMovieById = React.useMemo(() => {
-       return MOVIES_LIST.find(item => item.id === activeMovie);
+       return moviesList?.find(item => item.id === activeMovie);
```
Comment (mmarkelov) [inline src/components/Header/Header.tsx]: I am not sure that you need `?` here. `moviesList` is empty array by default

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -46,7 +57,19 @@ export const Header = () => {
                         </>
                 }
             </div>
-            {activeMovie && getMovieById ? <MovieDetails {...getMovieById} /> : <SearchBlock />}
+            {
+                activeMovie && getMovieById
```
Comment (mmarkelov) [inline src/components/Header/Header.tsx]: `getMovieById` looks like a function. But it’s movie object

File: src/components/Menu/Menu.tsx [hunk 3]
```diff
@@ -29,23 +33,32 @@ export const Menu = ({closeMenu}:MenuProps) => {
                 />
                 <Modal
                     open={openEditModal}
-                    onClose={() => setOpenEditModal(false)}
+                    onClose={() => {
+                        setOpenEditModal(false);
+                        closeMenu();
+                    }}
```
Comment (mmarkelov) [inline src/components/Menu/Menu.tsx]: You can create `onCloseModal` function to reuse it.
```js
const onCloseModal = () => {
    setOpenEditModal(false);
    closeMenu();
}
```

File: src/components/MovieContainer/MovieContainer.tsx [hunk 4]
```diff
@@ -2,14 +2,28 @@ import React from 'react';
...
 
 import styles from './MovieContainer.module.css';
 
 export const MovieContainer = () => {
+    const dispatch = useDispatch();
+    const {filter, sort, search, setFilter, setSort} = useNonNullContext(FiltersContext);
+
+    React.useEffect(() => {
+        getMovies(dispatch, {filter, sort, search});
+    }, [sort, filter]);
```
Comment (mmarkelov) [inline src/components/MovieContainer/MovieContainer.tsx]: You passed `search` in `getMovies`. Shouldn't it be in dependencies array?

File: src/components/MovieDetails/MovieDetails.tsx [hunk 5]
```diff
@@ -3,38 +3,38 @@ import React from 'react';
...
+    tagline: string;
+    vote: number;
+    date: string;
+    runtime: number;
+    overview: string;
 }
 
 export const MovieDetails = ({
-    image,
+    poster ,
```
Comment (mmarkelov) [inline src/components/MovieDetails/MovieDetails.tsx]: `poster ,` -> `poster,`

File: src/components/MovieList/MovieList.tsx [hunk 6]
```diff
@@ -1,22 +1,25 @@
...
+import {MovieState} from '../../redux/reducer';
+import {RootReducer} from '../../store';
 
 export const MovieList = () => {
+    const {moviesList} = useSelector<RootReducer, MovieState>((state) => state.movies);
+
     return (
         <div className={styles.list}>
-            {MOVIES.map((movie) => (
+            {moviesList?.map((movie) =>
```
Comment (mmarkelov) [inline src/components/MovieList/MovieList.tsx]: `?` is unnecessary here

File: src/redux/action.ts [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        title: title,
+        release_date: date,
+        poster_path: urlMovie,
+        genres: genre,
+        overview: overview,
+        runtime: Number(runtime),
+    });
+    fetch(url, {
+        method: 'POST',
+        headers: headers,
```
Comment (mmarkelov) [inline src/redux/action.ts]: `headers: headers,` -> `headers,`

File: src/redux/action.ts [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+    title,
+    date,
+    urlMovie,
+    genre,
+    overview,
+    runtime
+}: FormValues) => {
+    let url = `http://localhost:4000/movies`;
+    const data = JSON.stringify({
+        title: title,
```
Comment (mmarkelov) [inline src/redux/action.ts]: `title: title,` -> `title,`

File: src/redux/action.ts [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+        release_date: date,
+        poster_path: urlMovie,
+        genres: genre,
+        overview: overview,
```
Comment (mmarkelov) [inline src/redux/action.ts]: `overview,`

File: src/redux/action.ts [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+    date,
+    urlMovie,
+    genre,
+    overview,
+    runtime,
+    id
+}: FormValues) => {
+    let url = `http://localhost:4000/movies`;
+    const data = JSON.stringify({
+        title: title,
```
Comment (mmarkelov) [inline src/redux/action.ts]: `title,`

File: src/redux/action.ts [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+        release_date: date,
+        poster_path: urlMovie,
+        genres: genre,
+        overview: overview,
+        runtime: runtime,
+        id: id,
```
Comment (mmarkelov) [inline src/redux/action.ts]:
```js
overview,
runtime,
id,
```

File: src/redux/action.ts [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+    });
+
+    fetch(url, {
+        method: 'PUT',
+        headers: headers,
```
Comment (mmarkelov) [inline src/redux/action.ts]: `headers,`

File: src/redux/reducer.ts [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+export interface MovieState {
+    moviesList: Movie[];
+}
+
+export const MovieReducer = (state = initialState, action: ActionType): MovieState => {
+    const deleteItem = (list, id) => {
+        if(list) {
+            return list.filter((movie) => movie.id !== id);
+        }
+        return list;
```
Comment (mmarkelov) [inline src/redux/reducer.ts]: Also not sure that you need this check here. But if it’s necessary you can write it like this:
```js
return list?.filter((movie) => movie.id !== id);
```

File: src/redux/reducer.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+            };
+        case CREATE_MOVIES:
+            return {
+                ...state,
+                moviesList: [...state.moviesList, ...action.payload],
+            };
+        case UPDATE_MOVIES:
+            return {
+                ...state,
+                moviesList: editItem(state.moviesList, action.payload[0]),
```
Comment (mmarkelov) [inline src/redux/reducer.ts:39]: I'm not sure that you always need to change the first movie

Comment (DariaGunina) [reply src/redux/reducer.ts:39]: Thanks for the comment. In this case, an array comes in and one movie is updated

**Pull request merged**